relay-server/src/envelope.rs

```diff
@@ -550,7 +550,11 @@ impl Item {
             ItemType::Metrics | ItemType::MetricBuckets => None,
             ItemType::FormData => None,
             ItemType::UserReport => None,
-            ItemType::Profile => Some(DataCategory::Profile),
+            ItemType::Profile => Some(if indexed {
+                DataCategory::ProfileIndexed
+            } else {
+                DataCategory::Profile
+            }),
```

Comment:
`indexed` depends on whether metrics were extracted from a transaction. A profile counts towards `Profile` until metrics are extracted from the corresponding transaction. After that, it counts as `ProfileIndexed`.